core/trino-main/src/main/java/io/trino/server/HttpRequestSessionContextFactory.java

```diff
@@ -391,6 +389,15 @@ private static ResourceEstimates parseResourceEstimate(ProtocolHeaders protocolH
         return builder.build();
     }
 
+    private static List<String> splitSessionPropertyValue(String value)
+    {
+        int lastDot = value.lastIndexOf('.');
+        if (lastDot == -1) {
+            return ImmutableList.of(value);
+        }
+        return ImmutableList.of(value.substring(0, lastDot), value.substring(lastDot + 1));
```

Comment:
Consider adding `private record ParsedSessionPropertyName(Optional<String> catalog, String name)` for the response of this.  That will make the code above much simpler.